src/main/java/taskmate/tools/TaskList.java

```diff
@@ -0,0 +1,145 @@
+package taskmate.tools;
+
+import taskmate.exceptions.NoDataException;
+import taskmate.exceptions.TaskNotFoundException;
+import taskmate.tools.tasks.Deadline;
+import taskmate.tools.tasks.Event;
+import taskmate.tools.tasks.Task;
+import taskmate.tools.tasks.Todo;
+
+import java.util.ArrayList;
+
+public class TaskList {
+
+    ArrayList<Task> tasks = new ArrayList<Task>();
+    int numTotalTasks = 0;
+    int numIncompleteTasks = 0;
+
+    public TaskList() {
+    }
+
+    public TaskList(String contentsFromDisk) throws NoDataException {
+        if (contentsFromDisk.isEmpty()) {
+            throw new NoDataException();
+        }
+
+        String[] lines = contentsFromDisk.split("\\n");
+        String taskType, name, by, from, to, delimiter, delimiter2;
```

Comment:
Maybe you can make it clearer by specify the name of the dilimiter instead of calling it delimiter2?